### PR TITLE
Replace Win32API with Fiddle, update appveyor.yml

### PR DESCRIPTION
**Pull request by MSP-Greg**
My fork's Appveyor build passed and is at:
https://ci.appveyor.com/project/MSP-Greg/mysql2/builds/25306388

### CODE REVIEW

**Comment by sodabrew** [comment]
I remember doing it this way because Win32API is in core. Is it deprecated/removed and is Fiddle bundled with core now? At which Ruby does this change? (In other words what revisions are being left behind?)

**Comment by MSP-Greg** [comment]
> I remember doing it this way because Win32API is in core

I think it's always been part of the std-lib.

>Is it deprecated/removed

See [`ext/win32/lib/Win32API.rb:5`](https://github.com/ruby/ruby/blob/11f8c89171ff0b260891dec064cf69d9400a0704/ext/win32/lib/Win32API.rb#L5).  I haven't gone all the way back, but also see https://github.com/ruby/ruby/commit/dd3aec25d63e865ce0e90b2ff870d7bcc39b1a56.

Then again, I've joked about it, given that it's been deprecated for a very long time.

Win32API uses Fiddle.  Really, I'm just tired of seeing warning messages...

File: lib/mysql2.rb [hunk 1]
```diff
@@ -20,9 +20,12 @@
   end
 
   if dll_path
-    require 'Win32API'
-    LoadLibrary = Win32API.new('Kernel32', 'LoadLibrary', ['P'], 'I')
-    if LoadLibrary.call(dll_path).zero?
+    require 'fiddle'
+    kernel32 = Fiddle.dlopen 'kernel32'
```
Comment (sodabrew) [inline lib/mysql2.rb]: I think this should still be initial cap `Kernel32` ?

**Comment by sodabrew** [comment]
Looks like Fiddle is core as far back as Ruby 2.0: https://ruby-doc.org/stdlib-2.0.0/libdoc/fiddle/rdoc/Fiddle.html

Meanwhile, I knew there was some history in my memory banks around Win32API: https://github.com/ruby/ruby/pull/835

See also https://github.com/brianmario/mysql2/issues/584

Since the issue came up in the mysql2 0.3.x series, continuing to use `Win32API` must have been to continue supporting Ruby 1.8/1.9 (without Fiddle in core). Since those versions are no longer supported I see no reason not to switch.

_I am reminded to get a release out..._

File: lib/mysql2.rb (hunk 1 above)
Comment (MSP-Greg) [reply lib/mysql2.rb]: Umm, not reusing it are we?

**Comment by MSP-Greg** [comment]
Note to self - ALWAYS USE 'W' VERSIONS...

Some users may have UTF-8 characters in their user names...  Updated.  [Passing Appveyor build](https://ci.appveyor.com/project/MSP-Greg/mysql2/builds/25343815)

File: lib/mysql2.rb (hunk 1 above)
Comment (sodabrew) [reply lib/mysql2.rb]: I mean the library:`Win32API.new('Kernel32', ...`
vs. `Fiddle.dlopen 'kernel32'`

Comment (MSP-Greg) [reply lib/mysql2.rb]: Sorry. between multi-tasking & bouncing between languages, I often miss the obvious.

Actually, the file is `kernel32.dll`, and as you know, Windows file system is case-insensitive (which can cause issues when comparing paths, use `casecmp`...)

Also, ruby/reline essentially created its own `Win32API` class to avoid the deprecation warnings, and it's all lower case.  See [lib/reline/windows.rb
](https://github.com/ruby/ruby/blob/9b18cb341f08c725d74f667ef3724c4cc852fdbd/lib/reline/windows.rb#L55-L62)

Long story short, I think we're ok...
